util/types/time.go

```diff
@@ -548,37 +561,24 @@ func parseDatetime(str string, fsp int) (Time, error) {
 		err error
 	)
 
-	seps := parseDateFormat(str)
+	seps, fracStr := splitDateTime(str)
 	switch len(seps) {
 	case 1:
+		sep := seps[0]
 		// No delimiter.
-		if len(str) == 14 {
+		if len(sep) == 14 {
 			// YYYYMMDDHHMMSS
-			_, err = fmt.Sscanf(str, "%4d%2d%2d%2d%2d%2d", &year, &month, &day, &hour, &minute, &second)
-		} else if len(str) == 12 {
+			_, err = fmt.Sscanf(sep, "%4d%2d%2d%2d%2d%2d", &year, &month, &day, &hour, &minute, &second)
+		} else if len(sep) == 12 {
 			// YYMMDDHHMMSS
-			_, err = fmt.Sscanf(str, "%2d%2d%2d%2d%2d%2d", &year, &month, &day, &hour, &minute, &second)
+			_, err = fmt.Sscanf(sep, "%2d%2d%2d%2d%2d%2d", &year, &month, &day, &hour, &minute, &second)
 			year = adjustYear(year)
-		} else if len(str) == 8 {
+		} else if len(sep) == 8 {
 			// YYYYMMDD
-			_, err = fmt.Sscanf(str, "%4d%2d%2d", &year, &month, &day)
-		} else if len(str) == 6 {
+			_, err = fmt.Sscanf(sep, "%4d%2d%2d", &year, &month, &day)
+		} else if len(sep) == 6 {
 			// YYMMDD
-			_, err = fmt.Sscanf(str, "%2d%2d%2d", &year, &month, &day)
-			year = adjustYear(year)
-		} else {
-			return ZeroDatetime, errors.Trace(ErrInvalidTimeFormat)
-		}
-	case 2:
-		s := seps[0]
-		fracStr = seps[1]
-
-		if len(s) == 14 {
-			// YYYYMMDDHHMMSS.fraction
```

Comment:
no need to hanlde .fraction, it was split to fracStr, and handle it independently